Build failure Windows GHC 7.8 32bit

Using the given dlls package, the build fails with the following:
```
[33 of 33] Compiling Numeric.LinearAlgebra.Static ( src\Numeric\LinearAlgebra\Static.hs, dist\dist-sandbox-3695bf19\build\Numeric\LinearAlgebra\Static.o )

src\C\vector-aux.c:766:30:
     warning: 'struct random_data' declared inside parameter list

src\C\vector-aux.c:766:30:
     warning: its scope is only this definition or declaration, which is probably not what you want
src\C\vector-aux.c: In function 'urandom':

src\C\vector-aux.c:768:5:
     warning: implicit declaration of function 'random_r'
src\C\vector-aux.c: At top level:

src\C\vector-aux.c:775:18:
     warning: 'struct random_data' declared inside parameter list
src\C\vector-aux.c: In function 'gaussrand':

src\C\vector-aux.c:782:13:
     warning: passing argument 1 of 'urandom' from incompatible pointer type

src\C\vector-aux.c:766:15:
     note: expected 'struct random_data *' but argument is of type 'struct random_data *'

src\C\vector-aux.c:783:4:
     warning: passing argument 1 of 'urandom' from incompatible pointer type

src\C\vector-aux.c:766:15:
     note: expected 'struct random_data *' but argument is of type 'struct random_data *'
src\C\vector-aux.c: In function 'random_vector':

src\C\vector-aux.c:802:24:
     error: storage size of 'buffer' isn't known

src\C\vector-aux.c:804:31:
     error: invalid application of 'sizeof' to incomplete type 'struct random_data'

src\C\vector-aux.c:807:5:
     warning: implicit declaration of function 'initstate_r'

src\C\vector-aux.c:802:24:  warning: unused variable 'buffer'
```

If I change that line in `vector-aux.c` to `#include "windows_random_r.h"` it works with this warning:

````
In file included from src\Internal\C\vector-aux.c:10:0:

src\Internal\C\windows_random_r.h:8:0:  warning: "NULL" redefined
````

but when I go to load GHCi:

````
ghc.exe: C:\Users\Douglas\Documents\GitHub\dimensional-dk-linalg\.cabal-sandbox\i386-windows-ghc-7.10.1\hmatr_6XugfZZq0AhJEWyCsjXSV1\HShmatrix-0.17.0.0-6XugfZZq
0AhJEWyCsjXSV1.o: unknown symbol `___muldc3'
ghc.exe: unable to load package `hmatrix-0.17.0.0'
````

Now in a clean sandbox I get:
````
Installed vector-0.11.0.0
Configuring hmatrix-0.17.0.0...
Building hmatrix-0.17.0.0...
Preprocessing library hmatrix-0.17.0.0...

In file included from src\Internal\C\vector-aux.c:10:0:

src\Internal\C\windows_random_r.h:8:0:  warning: "NULL" redefined
C:/Program Files (x86)/MinGHC-7.10.1/ghc-7.10.1/mingw/bin/../lib/gcc/mingw32/4.5.2/include/stddef.h:400:0: note: this is the location of the previous definition

<command line>: does not exist: src\C\windows_random_r.c
Failed to install hmatrix-0.17.0.0
````

--- a/INSTALL.md
+++ b/INSTALL.md
@@ -61,7 +61,9 @@ using this method.
 
 2) Download and unzip somewhere OpenBLAS http://www.openblas.net/
 
-3) > cabal install --flags=openblas --extra-lib-dirs=C:\...\OpenBLAS\lib --extra-include-dir=C:\...\OpenBLAS\include
+3) In a normal Windows cmd:
+
+     > cabal install --flags=openblas --extra-lib-dirs=C:\...\OpenBLAS\lib --extra-include-dir=C:\...\OpenBLAS\include
 
 ## Tests ###############################################
 

--- a/INSTALL.md
+++ b/INSTALL.md
@@ -56,8 +56,11 @@ using this method.
 [winpack]: https://github.com/downloads/AlbertoRuiz/hmatrix/gsl-lapack-windows.zip
 
 ### Alternative Windows build
+
 1) > cabal update
+
 2) Download and unzip somewhere OpenBLAS http://www.openblas.net/
+
 3) > cabal install --flags=openblas --extra-lib-dirs=C:\...\OpenBLAS\lib --extra-include-dir=C:\...\OpenBLAS\include
 
 ## Tests ###############################################

--- a/packages/base/hmatrix.cabal
+++ b/packages/base/hmatrix.cabal
@@ -1,5 +1,5 @@
 Name:               hmatrix
-Version:            0.17.0.0
+Version:            0.17.0.1
 License:            BSD3
 License-file:       LICENSE
 Author:             Alberto Ruiz
@@ -74,7 +74,7 @@ library
                         Internal.Util
                         Internal.Modular
                         Internal.Static
-                        
+
     C-sources:          src/Internal/C/lapack-aux.c
                         src/Internal/C/vector-aux.c
 

--- a/packages/base/src/Internal/C/vector-aux.c
+++ b/packages/base/src/Internal/C/vector-aux.c
@@ -932,14 +932,19 @@ int vectorScan(char * file, int* n, double**pp){
 
 ////////////////////////////////////////////////////////////////////////////////
 
-#if defined (__APPLE__) || (__FreeBSD__) || defined(_WIN32) || defined(WIN32)
+#if defined (__APPLE__) || (__FreeBSD__)
 /* FreeBSD and Mac OS X do not provide random_r(), thread safety cannot be
    guaranteed.
    For FreeBSD and Mac OS X, nrand48() is much better than random().
    See: http://www.evanjones.ca/random-thread-safe.html
 */
 #pragma message "randomVector is not thread-safe in OSX and FreeBSD"
+#endif
 
+#if defined (__APPLE__) || (__FreeBSD__) || defined(_WIN32) || defined(WIN32)
+/* Windows use thread-safe random
+   See: http://stackoverflow.com/questions/143108/is-windows-rand-s-thread-safe
+*/
 inline double urandom() {
     /* the probalility of matching will be theoretically p^3(in fact, it is not)
        p is matching probalility of random().

--- a/packages/base/src/Internal/C/vector-aux.c
+++ b/packages/base/src/Internal/C/vector-aux.c
@@ -932,7 +932,7 @@ int vectorScan(char * file, int* n, double**pp){
 
 ////////////////////////////////////////////////////////////////////////////////
 
-#if defined (__APPLE__) || (__FreeBSD__)
+#if defined (__APPLE__) || (__FreeBSD__) || defined(_WIN32) || defined(WIN32)
 /* FreeBSD and Mac OS X do not provide random_r(), thread safety cannot be
    guaranteed.
    For FreeBSD and Mac OS X, nrand48() is much better than random().

